push.gradle

```diff
@@ -0,0 +1,34 @@
+import java.util.regex.Pattern
+
+def doExtractStringFromManifest(name) {
+    def manifestFile = file("AndroidManifest.xml")
+    if (!manifestFile.exists()) {
+      manifestFile = file("src/main/AndroidManifest.xml")
+    }
+    def pattern = Pattern.compile(name + "=\"(.*?)\"")
+    def matcher = pattern.matcher(manifestFile.getText())
+    matcher.find()
+    return matcher.group(1)
+}
+
+android {
+    defaultConfig {
+        multiDexEnabled true
+        applicationId = doExtractStringFromManifest("package")
```

Comment:
This breaks/conflicts with a custom-set `applicationId` from the user (at the `/platforms/android` level). Do we really need to do this? What's the goal of this `push.gradle` file?

Reply:
Why does this "break/conflict"?
Yes we need this to improve the build speed.

Reply:
Even though the default value for `applicationId` is the name of the package from AndroidManifest, it won't always/necessarily be that. One of my applications, for example, keeps a constant value for the package, while has multiple values for the `applicationId` so that I can maintain custom versions of the same application. I haven't tested it yet on my case, but I'd be willing to say that could conflict with scenarios like mine where the `applicationId` is customized. I could be wrong, but it's definitely a danger-sign for me.

On another topic, how exactly does this gradle file improve build speed?

Reply:
We're extracting the android package name from the Manifest file, so in all the cases this should always be the same (each time the file should be updated by cordova files generation), In your case you can change only the package name globally and It should be working 👍.
The android build is based on dex, so the multidex enabled should improve the build speed ✌️

Reply:
The `applicationId` is set since cordova-android 5.1.0, so you don't really need to set the value based on the AndroidManifest.xml value, it should always match now. I recently removed this code in https://github.com/phonegap/phonegap-plugin-push/pull/2367
And the multidex is still applied, but by using a dependency plugin https://github.com/phonegap/phonegap-plugin-push/blob/master/plugin.xml#L47

So this push.gradle file is not needed